src/main/java/seedu/duke/commands/FindCommand.java

```diff
@@ -13,20 +22,116 @@ public class FindCommand extends Command {
     String term;
     String[] flags;
 
+    /**
+     * Initializes FindCommand object
+     * with term and flags obtained from the arguments provided
+     * @param arguments
+     */
     public FindCommand(String arguments) {
         String[] details = arguments.split(FLAG_SEPARATOR);
         this.term = details[0];
         this.flags = Arrays.copyOfRange(details, 1, details.length);
     }
 
+    /**
+     * Finds products with a given term and filters them
+     * according to given flag values.
+     *
+     * Adds a foodItem to the result
+     * only if each flag was satisfied and the loop iteration
+     * wasn't interrupted early by non-matching flag value
+     *
+     * @param foodList
+     * @return CommandResult
+     * @throws DukeException
+     */
     @Override
     public CommandResult execute(FoodList foodList) throws DukeException {
-        FoodList result = foodList.findFood(term, flags);
+        FoodList result = new FoodList();
+        if(Objects.equals(term, "") && flags.length == 0) {
+            return new CommandResult("No food found for such query");
+        }
+
+        foodItemLoop:
+        for (Food foodItem : foodList.getFoodList()) {
+            String foodItemName = foodItem.getName();
+            LocalDate expiryDate = foodItem.parseExpiryDate();
+            String foodCategory = foodItem.getCategoryString(foodItem.getCategory());
+            String foodUnit = foodItem.getUnit();
+            boolean hasTerm = foodItemName.toLowerCase().contains(term.toLowerCase().trim());
+
+            if (hasTerm) {
+                for (String flag : flags) {

```

Comment:
good way of using flags to determine what parameters the user want to update. This way allow multiple parameters to be edited at the same time in one command.